joblib/test/test_numpy_pickle.py

```diff
@@ -916,3 +929,90 @@ def test_load_memmap_with_big_offset(tmpdir):
     np.testing.assert_array_equal(obj, memmaps)
+
+
+class BinaryCompressorTestFile(io.BufferedIOBase):
+    pass
+
+
+def test_register_compressor(tmpdir):
+    # Check that registering compressor file works.
+    compressor_name = 'test-name'
+    compressor_prefix = 'test-prefix'
+
+    register_compressor(compressor_name, compressor_prefix,
+                        BinaryCompressorTestFile)
+
+    assert (_EXTRA_COMPRESSORS[compressor_name].object ==
+            BinaryCompressorTestFile)
+    assert _EXTRA_COMPRESSORS[compressor_name].prefix == compressor_prefix
+
+    # Remove this dummy compressor file from extra compressors because other
+    # tests might fail because of this
+    _EXTRA_COMPRESSORS.pop(compressor_name)
+
+
+@parametrize('invalid_name', [1, (), {}])
+def test_register_compressor_invalid_name(invalid_name):
+    # Test that registering an invalid compressor name is not allowed.
+    with raises(ValueError) as excinfo:
+        register_compressor(invalid_name, b'prefix', None)
+    excinfo.match("Compressor name should be a string")
+
+
+class InvalidFileObject():
+    pass
+
+
+@parametrize('invalid_fileobj', [None, InvalidFileObject])
+def test_register_compressor_invalid_fileobj(invalid_fileobj):
+    # Test that registering an invalid file object is not allowed.
+    with raises(ValueError) as excinfo:
+        register_compressor('invalid', b'prefix', invalid_fileobj)
+    excinfo.match("Compressor should implement the file object interface")
+
+
+def test_register_compressor_already_registered():
+    # Test registration of existing compressor files.
+    compressor_name = 'gzip'
+    with raises(ValueError) as excinfo:
+        register_compressor(compressor_name, b'prefix', BinaryZlibFile)
+    excinfo.match("Compressor '{}'already registered.".format(compressor_name))
+
+    register_compressor('gzip', b'prefix', gzip.GzipFile, force=True)
+
+    assert compressor_name in _EXTRA_COMPRESSORS
+    assert _EXTRA_COMPRESSORS[compressor_name].object == gzip.GzipFile
+
+    # Remove this dummy compressor file from extra compressors because other
+    # tests might fail because of this
+    _EXTRA_COMPRESSORS.pop(compressor_name)
+
+
+@with_lz4
+def test_lz4_compression(tmpdir):
+    # Check that lz4 can be used when dependency is available.
+    import lz4.frame
+    compressor = 'lz4'
+    assert compressor in _COMPRESSORS
+    assert compressor in _EXTRA_COMPRESSORS
+    assert _EXTRA_COMPRESSORS[compressor].object == lz4.frame.LZ4FrameFile
+
+    fname = tmpdir.join('test.lz4').strpath
+    data = 'test data'
+    numpy_pickle.dump(data, fname, compress=compressor)
+
+    assert open(fname, 'rb').read(len(_LZ4_PREFIX)) == _LZ4_PREFIX
+    assert numpy_pickle.load(fname) == data
+
+
+@without_lz4
+def test_lz4_compression_without_lz4(tmpdir):
+    # Check that lz4 cannot be used when dependency is not available.
+    fname = tmpdir.join('test.no.lz4').strpath
+    data = 'test data'
+    compressor = 'lz4'
```

Comment:
The test would be more readable by not defining this local variable and using the literatl `compressor='lz4'` directly in the code.